src/PrestaShopBundle/Service/Routing/Router.php

```diff
@@ -47,6 +55,12 @@ class Router extends BaseRouter
     public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
     {
         $url = parent::generate($name, $parameters, $referenceType);
+
+        // Don't tokenize URLs for routes that are prefixed with "_"
```

Comment:
It's a bit "convention-based" magic 🤔
Maybe we could make it a bit more configuration-oriented, to avoid ambiguous behavior that might create buggy behaviors and developer habits ? For example maybe routes can be tagged (using the `_defaults` maybe) ?

Reply:
it's not something new introduced by this PR, it's actually behaving like this since a few years back... So it would break BC if I change it.

Currently on develop it's not checking token on routes, which are prefixed by `_`: https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/EventListener/TokenizedUrlsListener.php#L89

This change only fixes the URL generation in router, to make it consistent with tokenization listener.

Reply:
This gives me chills... 😱